windowing.py

```diff
@@ -1,5 +1,48 @@
 import numpy as np
 from skimage.util.shape import view_as_windows
+from dataclasses import dataclass
+
+@dataclass
+class Initial_Filtering_Params:
+    window_size: int
+    stride: int
+    divisor: int
+    rows: int 
+
+@dataclass
+class Precise_Filtering_Params:
+    window_size: int
+    stride: int
+    threshold: int
+    rows: int
```

Comment:
Can you use the same data class for both?

Reply:
I could I just thought it might be make it more readable for the user to make it clear you need 2 classes and what those values actually affect

Reply:
I think it would be more clear if you used the same class for both and just named the instance differently? They both have specific parameters like threshold and divisor that only come into play for different points during the processing but it seems like it could be a generic container. Do you think it would still be confusing for the user?